src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AnonymousElectorDocumentService.kt

```diff
@@ -60,11 +60,19 @@ class AnonymousElectorDocumentService(
     @Transactional
     fun updateAnonymousElectorDocument(eroId: String, updateAedDto: UpdateAnonymousElectorDocumentDto) {
         with(updateAedDto) {
-            val anonymousElectorDocuments = getAnonymousElectorDocuments(eroId, sourceReference)
+            val gssCodes = eroService.lookupGssCodesForEro(eroId)
+            val anonymousElectorDocuments = getAnonymousElectorDocumentsSortedByDate(gssCodes, sourceReference)
             if (anonymousElectorDocuments.isEmpty()) {
                 throw CertificateNotFoundException(eroId, ANONYMOUS_ELECTOR_DOCUMENT, sourceReference)
             }
-            // TODO EIP1-5925
+            anonymousElectorDocuments.forEach {
+                if (valueHasChanged(email, it.contactDetails!!.email)) {
```

Comment:
Can we not just update an email and phone for the latest record?

Reply:
We think it's better to update all the AEDs. The contact details do not really belong to an AED per se (though they've been modelled that way) and I can't see any benefit in keeping out of date details.

Reply:
If we do want to update all the records, then does it make sense to check only the latest record and see if it has changed, and if yes then we change all email records (same for phone)?

Reply:
I'm not so sure to be honest. Realistically there will usually only be one and at most two or three(?). Once they're retrieved from the DB, iterating over them is negligible in terms of performance and I still think it's best to check them all.